config.js

```diff
@@ -88,6 +88,16 @@ module.exports = {
     // this will also cause sampling decisions made by other distributed components to be
     // ignored. This is useful for aggregating traces generated by different cloud platform
     // projects.
-    ignoreContextHeader: false
+    ignoreContextHeader: false,
+
+    // A path to a key file relative to the current working directory. If this
+    // field is set, the contents of the pointed file will be used for
+    // authentication instead of your application default credentials.
+    keyFilename: null,
+
+    // The contents of a key file. If this field is set, its contents will be
+    // used for authentication instead of your application default credentials.
+    // If keyFilename is also set, the value of credentials will be ignored.
+    credentials: null
```

Comment:
Perhaps it could be easier for users if there was just a single `credentials` attribute.  If the value of the attribute was a file path, it would be assumed to the filename to the key file.  Otherwise, the value would be assumed to be the contents of the key file.

The advantage is that there is one attribute that specifies credentials so that if the user encounters problems, there are less interrelated attributes to consider, and the single attribute handles priority.

The disadvantage is that a single attribute no longer has a single interpretation.

Just an idea.